src/auth/service.ts

```diff
@@ -228,7 +230,12 @@ export class AuthenticationService {
    */
   login(login: string, pwd: string, otp: string = undefined): Promise<void | User> {
     const url = `${this.baseUrl_}/${RouteSuffix.LOGIN}`;
-    const params = {'login': login, 'password': pwd};
+    const params = {
+      'login': login,
+      'password': pwd,
+      came_from:
+        user.getLoginMessage().value == loginMessageRequired ? this.originalUrl : window.location.href,
```

Comment:
Tell me if I'm wrong, but does this.originalUrl can change somewhere else ? As I understand it is initialized as window.location.href. 

If the condition is true or false, the end result is window.location.href from this.originalUrl or window.location.href so the same value ?

Reply:
I don't know if the commant will explain better than the full answer... :-)

Reply:
I guess the full answer explains it.

Now the comment you wrote, not so sure. :-)

Reply:
And with this comment?

Reply:
Good, thanks !